src/Builder/Visitor/AddPluginToPluginListVisitor.php

```diff
@@ -159,6 +167,7 @@ protected function createArrayItemWithInstanceOf(): ArrayItem
             (new BuilderFactory())->new(
                 (new ClassHelper())->getShortClassName($this->className),
             ),
+            $this->index,
```

Comment:
I would build the stmt inside of the visitor and not outside. What do you think?